src/service/navigation.js

```diff
@@ -317,6 +317,7 @@ export class Navigation {
       // TODO(alabiaga): investigate fix for handling A2A and custom link
       // protocols.
       this.expandVarsForAnchor_(target);
+      this.anchorMutatorHandlers_(target, e);
```

Comment:
Would be nicer and more consistent if grouped into a `handleContextClick_` method.